src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -269,39 +269,45 @@ CPP_template(typename NaryOperation)(
     const SparqlExpression* child0 = this->getChildAtIndex(0).value();
     const SparqlExpression* child1 = this->getChildAtIndex(1).value();
 
-    const auto getPrefilterExprVariableVec =
-        [](const SparqlExpression* child0, const SparqlExpression* child1,
-           bool startsWithVar) -> std::vector<PrefilterExprVariablePair> {
-      const auto* varExpr = dynamic_cast<const VariableExpression*>(child0);
-      if (!varExpr) {
-        return {};
-      }
+    const auto makePrefilterExpressionVec =
+        [](const std::pair<std::string, Variable>& strVarPair, bool mirrored) {
+          std::vector<PrefilterExprVariablePair> prefilterVec;
+          const auto& [prefixStr, variable] = strVarPair;
+          prefilterVec.emplace_back(
+              std::make_unique<prefilterExpressions::PrefixRegexExpression>(
+                  prefixStr, mirrored),
+              variable);
+          return prefilterVec;
+        };
 
-      const auto& optReferenceValue =
-          getIdOrLocalVocabEntryFromLiteralExpression(child1, true);
-      if (optReferenceValue.has_value()) {
-        return prefilterExpressions::detail::makePrefilterExpressionVec<
-            prefilterExpressions::CompOp::GE>(optReferenceValue.value(),
-                                              varExpr->value(), startsWithVar);
+    const auto getPrefixStrAndVariable = [](const SparqlExpression* child0,
+                                            const SparqlExpression* child1)
+        -> std::optional<std::pair<std::string, Variable>> {
+      const auto* varExpr = dynamic_cast<const VariableExpression*>(child0);
```

Comment:
I think we have a `getVariableOrNullopt` expression for this case (less low-levely thant the `dynamic_cast`.